doc/tools/docgram/README.md

```diff
@@ -0,0 +1,199 @@
+
+The grammar editing scripts `*.edit_mlg` are similar in format to `.mlg` files stripped
+of all OCaml features.  This is an easy way to include productions to match or add without
+writing another parser.  The "DOC_GRAMMAR" token at the beginning of each file
+signals the use of streamlined syntax.
+
+Each edit file has a series of items in the form of productions.  Items are applied
+in the order they appear.  There are two types of editing operations:
+
+* Global edits - edit rules that apply to the entire grammar in a single operation.
+These are identified by using specific reserved names as the non-terminal name.
+
+* Local edits - edit rules that apply to the productions of a single non-terminal.
+If the non-terminal name isn't reserved, the rule is a local edit.  Local edits
+that begin with a different set of reserved names edit existing productions.  For
+example `binders: [ | DELETE Pcoq.Constr.binders ]` deletes the production
+`binders: [ | Pcoq.Constr.binders]`
+
+Productions that don't begin with a reserved name are added to the grammar,
+such as `empty: [ | ]`, which adds a new non-terminal `empty` with an
+empty production on the right-hand side.
+
+Another example: `LEFTQMARK: [ | "?" ]` is a local edit that treats `LEFTQMARK` as
+the name of a non-terminal and adds one production for it.  (We know that LEFTQMARK
+is a token but doc_gram does not.)  `SPLICE: [ | LEFTQMARK ]` requests replacing all
+uses of `LEFTQMARK` anywhere in the grammar with its productions and removing the
+non-terminal.  The combined effect of these two is to replace all uses of
+`LEFTQMARK` to `"?"`.
+
+Here are the current operations.  They are likely to be refined as we learn
+what operations are most useful while we update the mlg files and documentation:
+
+Global edits
+------------
+
+DELETE - deletes the specified non-terminals anywhere in the grammar.
+Useful for removing non-terminals that only do lookahead that shouldn't be
+in the documentation.
+
+RENAME - each production specifies an (old name, new name) pair of
+non-terminals to rename.
+
+SPLICE - requests replacing all uses of the nonterminals anywhere in the
+grammar with its productions and removing the non-terminal.  (Doesn't work
+recursively; splicing for both `A: [ | B ]` and `B: [ | C ]` has to be
+done in 2 SPLICE operations.)
+
+EXPAND_LISTS - expands LIST0, LIST1, LIST* ... SEP and OPT constructs into
+new non-terminals
+
+Local edits
+-----------
+
+DELETE <production> - removes the specified production from the grammar
+
+EDIT <production> - modifies the specified production using the following tags
+that appear in the specified production:
+  USE_NT <name> LIST* - extracts LIST* as new nonterminal with the specified
+    new non-terminal name
+  ADD_OPT <grammar symbol> - looks for a production that matches the specified
+    production **without** <grammar_symbol>.  If found, both productions are
+    replaced with single production with `OPT <grammar_symbol>`
```

Comment:
Here as well, you should use bullet points so that this is properly rendered.

Furthermore, you should use more quoting (the entire document misses a lot of verbatim quoting). In particular, things such as `<name>` are interpreted as HTML tags and not displayed if you forget quoting. This make the document impossible to read.

Reply:
I usually only look at these files in the editor.  FWIW I notice that when I run `restview`, it doesn't include the `*.md` files when the parent directory is selected.  (And `restview` is what gave me the error for `:kbd:F12` in another PR.  I have V 2.9.2.)